include/Components/Button.hpp

```diff
@@ -0,0 +1,108 @@
+/*
+** Project FileShare, 2022
+**
+** Author Léo Lhuile
+**
+** Started on  Sat Jan 15 21:08:44 2022 Léo Lhuile
+** Last update Wed Sep 14 22:26:27 2022 Léo Lhuile
+**
+** Button.hpp : Class to manage a button with icon and extra informations.
+*/
+
+#pragma once
+
+#include <TGUI/TGUI.hpp>
+
+namespace FileShare::GUI::Components {
+    class Button : public tgui::BitmapButton {
+        public:
+            enum Alignment {
+                Left,
+                Center,
+                Right
+            };
+
+            enum Type {
+                Primary,
+                Secondary,
+                Soft,
+                Danger,
+            };
+
+            enum Size {
+                Small,
+                Normal,
+                Large
+            };
+
+            Button(const char* typeName = "Components::Button", bool initRenderer = true);
+            ~Button();
+
+            typedef std::shared_ptr<Button> Ptr;
+			typedef std::shared_ptr<const Button> ConstPtr;
+
+			static Button::Ptr create() { return std::make_shared<Button>(); }
+			static Button::Ptr copy(Button::ConstPtr widget) { return widget ? std::static_pointer_cast<Button>(widget->clone()) : nullptr; }
+
+            void setImageSize(float size) { this->imageSize = {size, size}; this->setImageScaling(-1); }
+            void setImageSize(const tgui::Vector2f &size) { this->imageSize = size; this->setImageScaling(-1); }
```

Comment:
```suggestion
            void setImageSize(float size) { this->setImageSize({size, size}); }
            void setImageSize(const tgui::Vector2f &size) { this->imageSize = size; this->setImageScaling(-1); }
```
Use overloaded functions to avoid repeating logic (currently repeating `setImageScaling`)